### DISCUSSION TITLE
Which identity provider to choose

### DISCUSSION THREAD

**Discussion by Jester0027** [bartosz-skejcik/devdeck#15] [Ideas]
Here are some options (non-exhaustive):
- [Supabase](https://supabase.com/auth)
Open-source, all-in-one backend solution that also happen to include an identity provider/authorization server.
Hosted [on-premise](https://supabase.com/docs/guides/self-hosting) or via [Supabase's cloud offers](https://supabase.com/pricing)
- [Auth0](https://auth0.com/)
Proprietary authorization server from Okta, cloud-based, free-tier available
- [Keycloak](https://www.keycloak.org/)
Open-source IDP & authorization server made by Red Hat, hosted on-premise
Extensible with "SPIs" (Service Provider Interfaces)
- [Pocketbase](https://pocketbase.io/)
Lightweight alternative to Firebase and Supabase, open-source, single executable that contains the whole server.
Contains an identity provider.
- Custom-built 😁
A lot of work but would have to advantage to control everything

**Comment by bartosz-skejcik** [comment]
I think we should go with supabase cause of the free self-hosting. i have a pretty god machine that would be perfect for that. and for development we can use their docker script to host an instance locally

https://supabase.com/docs/guides/self-hosting
